modules/distributor/distributor.go

```diff
@@ -398,15 +401,39 @@ func (d *Distributor) sendToIngestersViaBytes(ctx context.Context, userID string
 			return err
 		}
 
-		_, err = c.(tempopb.PusherClient).PushBytesV2(localCtx, &req)
+		response, err := c.(tempopb.PusherClient).PushBytesV2(localCtx, &req)
+		// response = {empty}, err = nil ==> entire batch processed successfully
+		// response = nil, err != nill ==> entire batch failed not related to livetraces or tracetoolong error
+		// response != {empty}, err = nil  ==>  at least one trace was discarded
```

Comment:
I feel like this would be easier to read as follows:

```suggestion
		response, err := c.(tempopb.PusherClient).PushBytesV2(localCtx, &req)
		metricIngesterAppends.WithLabelValues(ingester.Addr).Inc()

		if err != nil {
			// internal error, abandon the entire batch
			metricIngesterAppendFailures.WithLabelValues(ingester.Addr).Inc()
			return err
		}
		
		// batch was (partially) accepted, gather responses and we are done

		return nil
```

Reply:
ohh i like this